userspace/libsinsp/parsers.cpp

```diff
@@ -4921,6 +4921,37 @@ void sinsp_parser::parse_rw_exit(sinsp_evt *evt)
 					(*it)->on_read(evt, data, datalen);
 				}
 			}
+
+			//
+			// Check if recvmsg contains ancillary data. If so, we check for SCM_RIGHTS,
+			// which is used to pass FDs between processes, and update the sinsp state 
+			// accordingly via procfs scan.
+			//
+#ifndef _WIN32
+			if(etype == PPME_SOCKET_RECVMSG_X && evt->get_num_params() >= 5)
```

Comment:
SInce this logic is completely new, we have no other cases of single fd scan for one process, It would be amazing to check in a sinsp unit test if this logic really works...
One method could be to trigger a fake recvmsg event with the sinsp test engine on the init thread `tid:1` and check if the proc_scan is performed. WDYT?

Reply:
Let's give it a try!

Reply:
I have added that test!